Story/getting stared revision (#109)
vmware/vmware-aria-operations-integration-sdk: doc/mp-init.md

Path: my-adapter
```

#### Display Name
VMware Aria Operations Manager uses the _display name_ as the name of the Management Pack generated by this project. The name should include the name of the technology the Management Pack monitors.
```
Management pack display name: My Adapter
```

#### Adapter Key
The _adapter key_ is used internally to identify the Management Pack and Adapter. It should be unique and cannot contain spaces or other special characters. It also cannot start with a number. By default, it is set to the Display Name with special characters removed (if the Display Name begins with a number, 'Adapter' is prepended).
```
Management pack adapter key: MyAdapter
```

#### Description
A brief _description_ of the Management Pack and the technology it monitors. The description should include relevant versions of the monitored technology.
```
Management pack description: This management pack is an example for the VMware Aria Operations Integration SDK project.
```

#### Vendor
The name of the vendor/developer of the Management Pack. VMware Aria Operations Manager will display this information during the installation of the  Management Pack generated by this project.
```

> Management pack vendor: VMware

Reviewer: I'd suggest something like `MyCompany` here

Enter a path to the management pack icon file, or leave blank for no icon:
An icon can be added later by setting the 'pak_icon' key in 'manifest.txt' to the icon file name and adding the icon file to the root project directory.
```
[//]: # (TODO: Add this section back when we support it)
[//]: # (#### Language)
[//]: # (The _language_ for the Management Pack determines the language for the template source and build files.)
[//]: # (```)
[//]: # (Select a language for the adapter.)
[//]: # ( ❯ Python)
[//]: # (```)

##  Output
The init script creates a project in the given path with the following structure:
```
.
├── Dockerfile
├── adapter_requirements.txt
├── commands.cfg
├── eula.txt
├── manifest.txt
├── requirements.txt

> ├── venv

Reviewer: I think the `venv` folder name is now generated using the adapter name

├── app
│   ├── adapter.py
│   └── constants.py
├── conf
│   ├── describeSchema.xsd
│   └── resources
├── content
│   ├── alertdefs
│   │   └── alertDefinitionSchema.xsd
│   ├── customgroups
│   ├── dashboards
│   ├── files
│   │   ├── reskndmetric
│   │   ├── solutionconfig
│   │   ├── topowidget
│   │   └── txtwidget
│   ├── policies
│   ├── recommendations
│   ├── reports
│   ├── resources
│   ├── supermetrics
│   ├── symptomdefs
│   └── traversalspecs
│       └── traversalSpecsSchema.xsd
└── resources
    └── resources.properties
```

### Dockerfile (file)
Contains all necessary instructions to build a container with an HTTP server, the user's executable adapter
code, and any additional dependencies specified by the user.

### adapter_requirements.txt (file)

> This file defines all the dependencies needed by the adapter at run time. This file is copied onto the container where the adapter runs.

Reviewer: ```suggestion
This file defines all the dependencies needed by the adapter at run time. This file is copied onto the container where the adapter runs and used by the default Dockerfile to install dependencies into the container image.
```

### requirements.txt (file)

> This file defines all the dependencies needed for development of the adapter. This file does not get copied into the Dockerfile.

Reviewer: ```suggestion
This file defines all the dependencies needed for development of the adapter. This file does not get copied into the container image.
```

| license_type              | Management Packs made using the VMware Aria Operations Integration SDK do not support licensing.                                                                                                                                                                                                                                   |

<sup>1</sup> This property is required.

<sup>2</sup> Localized fields are specified in the property files located in the `resources/` directory. See [Adding Localization](adding_content.md#adding-localization).

### venv (directory)

A Python virtual environment for the project. The directory name is prefixed with __venv__ followed by the display name of the management pack. This virtual environment is created to allow users to use different versions of the SDK for every project.

**NOTE:** This directory is excluded from version control. Users cloning the project will have to create their own venv directory([ venv documentaion](https://docs.python.org/3/library/venv.html)).

> ### app (directory)

Reviewer: This is perhaps the most important directory, and should be at or near the top.

is running in, and can be used as a starting point for creating a new adapter. The template adapter uses [vmware-aria-operations-integration-sdk-lib](https://pypi.org/project/vmware-aria-operations-integration-sdk-lib/) to streamline the process of building adapter.
The template adapter has comments throughout the code to help new users understand the process of creating their own adapter using the existing code. For additional guidance creating adapters see our [Walkthroughs](../README.md#walkthroughs) section.

### conf (directory)
The conf directory contains configuration files used by VMware Aria Operations to create a management pack instance.

- describe.xml: An XML configuration file that defines the object model for an adapter, along with semantic definitions for use in data analysis and management.
- describeSchema.xsd: A file that defines what elements and attributes may appear in the `describe.xml` document. This file does not need to be included in a Management Pack but is useful for validating that the `describe.xml` file is error-free.

### content (directory)

> Contains all the content included in a management pack such as: alerts, dashboards, reports, recommendations, symptoms, and traversal specs. To learn more about adding content, consult the [Adding Content](adding_content.md) guidej

Reviewer: ```suggestion
Contains all the content included in a management pack such as: alerts, dashboards, reports, recommendations, symptoms, and traversal specs. To learn more about adding content, consult the [Adding Content](adding_content.md) guide.
```

### resources (directory)

> The resources directory contains localization file(s), by default `resources.properties`. This file specifies the mapping of `nameKey` attribute of elements inside the `conf/describe.xml`. For more information see [Adding Localization](adding_content.md#adding-localization).

Reviewer: The top-level 'resources' directory has the localization file(s) for the manifest.txt file, not describe. The describe's resources folder is in the 'conf' directory next to the describe.xml file itself.
